pkg/graph/resolvers/plan_discussion.go

```diff
@@ -12,7 +17,16 @@ import (
 )
 
 // CreatePlanDiscussion implements resolver logic to create a plan Discussion object
-func CreatePlanDiscussion(logger *zap.Logger, input *model.PlanDiscussionCreateInput, principal authentication.Principal, store *storage.Store) (*models.PlanDiscussion, error) {
+func CreatePlanDiscussion(
```

Comment:
@OddTomBrooks , @ClayBenson94 , I wanted to hear what you both think about removing some of these extra parameters, and instead wrapping more of these dependencies in context?
 
This new method signature requires context be passed, `logger` and `principal` already happen to live on the context. I think it could make sense to decorate the context with `email service` etc as well.

Reply:
That's an interesting wide-scale problem to solve. It's a balancing act. I'm not overly fond of the context pattern for passing services. As far as I can tell the primary use of context in Golang is to support preserving state in libraries. While it certainly can be used to transport services that is basically one step removed from the singleton or global pattern. It is possible to swap dependencies in the context stage but it is very limited and resource lifecycle is poorly managed.

I suspect it is an easier architectural question whether/how to group parameters into categorical objects (e.g. Services, CreatePlanDiscussionInput, etc). According to Clean Code this is a proper refinement. Another consideration is that functions with long parameter lists indicate multiple responsibility. Golang is more functional than I am accustomed, but one option may be to move resolver function groups into an OO model (PlanDiscussion struct) and use a factory pattern for dependency construction to simplify things.

Reply:
Agreed! It's outside of the scope of this PR. It does start getting into paradigm choice, it's something that's been floating around the back of my mind. Maybe these options are something we could explore during a pairing session sometime?

Those are some good points too about dependency swapping, the way we have it probably makes testing a good deal simpler as well 🤔. It might not be worth changing, but good to explore.